### PR TITLE
Move Code Generation From Bash to Swift Scripts

### PR DESCRIPTION
**Pull request by designatednerd**
This is explained in detail in #939, but at a basic level I'm sick of having so much untested bash code driving our code generation, so I'm moving a bunch of it to Swift.

### CODE REVIEW

**Comment by designatednerd** [comment]
OK folks, I would love some feedback on all this. Hit me with your best shots! 🥊

**Comment by sgoodwin** [comment]
Hi! I only looked at the docs part, but they look cool. Quite some setup, but then I could see this being long-term nice for a team (next time I'm on one doing GraphQL). Good idea!

**Comment by designatednerd** [comment]
Yes! A big piece of the idea here is to make it *way* more seamless for me to work piecemeal on some major code generation changes, which are all eventually going to live in the `ApolloCodegenLib` library.

**Comment by designatednerd** [comment]
(thought of a WAY easier way to get the sourceroot)

File: Sources/ApolloCodegenLib/Basher.swift [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import Foundation
+
+/// Bash command runner
+public struct Basher {
```
Comment (bleege) [inline Sources/ApolloCodegenLib/Basher.swift]: Great name!

**Comment by bleege** [comment]
+100 to moving away from Bash.  🎉

File: Sources/ApolloCodegenLib/Basher.swift (hunk 1 above)
Comment (designatednerd) [reply Sources/ApolloCodegenLib/Basher.swift]: 🔨 Thanks!

File: docs/source/api/ApolloCodegenLib/structs/ApolloCLI.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+```swift
+public static func createCLI(cliFolderURL: URL, timeout: Double) throws -> ApolloCLI
+```
+
+> Creates an instance of `ApolloCLI`, downloading and extracting if needed
+>
+> - Parameters:
+>   - cliFolderURL: The URL to the folder which contains the zip file with the CLI.
+>   - timeout: The maximum time which should be waited before indicating that the download timed out, in seconds.
```
Comment (StephenBarlow) [inline docs/source/api/ApolloCodegenLib/structs/ApolloCLI.md]: "which should be waited" -> "to wait"

Comment (StephenBarlow) [reply docs/source/api/ApolloCodegenLib/structs/ApolloCLI.md]: (Incorporate across all instances)

File: docs/source/api/ApolloCodegenLib/structs/ApolloCodegenOptions.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+>  - urlToSchemaFile: The URL to your schema file.
+>  - downloadTimeout: The maximum time which should be waited before indicating that the download timed out, in seconds. Defaults to 30 seconds.
+
+### `init(targetRootURL:downloadTimeout:)`
+
+```swift
+public init(targetRootURL folder: URL, downloadTimeout: Double = 30.0)
+```
+
+> Convenience initializer taking the root folder of a target and generate
```
Comment (StephenBarlow) [inline docs/source/api/ApolloCodegenLib/structs/ApolloCodegenOptions.md]: "taking" -> "that takes"
"generate" -> "generates"

**Comment by designatednerd** [comment]
It'll be a couple days before I ship this - I have some serious wrecking ball work to do with rearranging some code to get a couple things to work.

**Pull request merged**